### ISSUE TITLE
State Changes Do Not Rebuild `Select` Widget While Pop-up is Open

### ISSUE THREAD

**Issue by AndryHTC** [sunarya-thito/shadcn_flutter#79]
**Description**

When the state of the `Select` widget changes (e.g., adding new options dynamically), the UI does not reflect these changes in real-time if the pop-up is currently open. The children of the `Select` widget only rebuild when the pop-up is closed and reopened. This issue prevents dynamic updates from being visible to the user immediately.

**Example Code**

The following Dart code demonstrates the problem. A `Timer` adds a new option to the `Select` widget every 2 seconds. Although the `Ticker` logs indicate that options are being added, the UI does not update the `Select`'s children until the pop-up is closed and reopened.

```dart
import 'dart:async';
import 'package:flutter/material.dart';

class Test extends StatefulWidget {  
  const Test ({
    super.key,
  });

  @override
  State<Test> createState() => _TestState();
}

class _TestState extends State<Test> {
  List<String> filteredValues = ['Option 1', 'Option 2', 'Option 3'];
  String? selectedValue;
  Timer? _timer;
  int _optionCount = 3;

  @override
  void initState() {
    super.initState();
    // Start the ticker to add a new option every 2 seconds
    _timer = Timer.periodic(Duration(seconds: 2), (timer) {
      setState(() {
        _optionCount += 1;
        filteredValues.add('Option $_optionCount');
        debugPrint('Ticker added: Option $_optionCount');
      });
    });
  }

  @override
  void dispose() {
    _timer?.cancel(); // Cancel the timer when the widget is disposed
    super.dispose();
  }

  void _resetOptions() {
    setState(() {
      filteredValues = [];
      selectedValue = null;
      _optionCount = 0;
      debugPrint('Options reset');
    });
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        // Control Buttons
        Row(
          children: [
            ElevatedButton(
              onPressed: _resetOptions,
              child: Text('Reset'),
            ),
          ],
        ),
        SizedBox(height: 20),
        // Select Widget
        Select<String>(
          children: filteredValues.map((value) {
            return SelectItemButton<String>(
              value: value,
              child: Text(value),
            );
          }).toList(),
          emptyBuilder: (context) {
            return Text('No options available');
          },
          onChanged: (value) {
            setState(() {
              selectedValue = value;
              debugPrint('Selected: $value');
            });
          },
          itemBuilder: (BuildContext context, String item) {
            return Text(item);
          },
        ),
        SizedBox(height: 20),
        // Display Selected Value
        Text(
          selectedValue != null
              ? 'Selected: $selectedValue'
              : 'No option selected',
        ),
      ],
    );
  }
}
```

**Steps to Reproduce**

1. Integrate the above `Test` widget into your Flutter application.
2. Run the app and open the `Select` widget's pop-up.
3. Observe that every 2 seconds, a new option is added in the background (as indicated by the `Ticker` logs).
4. Notice that the new options do not appear in the pop-up until it is closed and reopened.

**Expected Behavior**

When new options are added to the `filteredValues` list, the `Select` widget's pop-up should automatically rebuild to display the updated list of options in real-time, even if the pop-up is currently open.

**Actual Behavior**

The `Select` widget's pop-up does not reflect the new options until it is closed and reopened. The `Ticker` confirms that options are being added, but the UI remains static while the pop-up is open.

**Possible Workaround**

Currently, the only workaround is to close and reopen the `Select` widget's pop-up to see the updated options.

**Additional Notes**

This issue affects the user experience by preventing real-time updates to the `Select` widget's options, which can be critical for dynamic filtering and selection scenarios.

---

**Comment by AmoonPod** [comment]
Can yoy please reply piece of shit?

**Comment by AndryHTC** [comment]
@sunarya-thito we love your package. Any ETA for this?

**Comment by sunarya-thito** [comment]
Probably around this week

**Comment by AmoonPod** [comment]
Any update on this? I'm waiting for the release.

**Comment by sunarya-thito** [comment]
I already released v0.0.21
https://pub.dev/packages/shadcn_flutter

**Comment by AmoonPod** [comment]
Yeah, I've seen it. But I was wondering when this issue is going to be resolved, since I'm having the same issue with v.0.0.21. 
Thank you!

**Comment by sunarya-thito** [comment]
Already fixed in 0.0.21
![test_widget_qv5QGqLTSz](https://github.com/user-attachments/assets/67bc6b63-afff-4744-b8c1-6135539089b6)
